pkg/platform/aks/aks.go

```diff
@@ -97,16 +102,35 @@ func init() { //nolint:gochecknoinits
 }
 
 // LoadConfig loads configuration values into the config struct from given HCL configuration.
-func (c *config) LoadConfig(configBody *hcl.Body, evalContext *hcl.EvalContext) hcl.Diagnostics {
-	if configBody == nil {
-		emptyConfig := hcl.EmptyBody()
-		configBody = &emptyConfig
+func (c *config) LoadConfig(cc *lkconfig.Config) hcl.Diagnostics {
```

Comment:
As we need to get rid of `LoadConfig()` completely, I'm not sure why we add more code into it now. Also accepting entire configuration here breaks the encapsulation :confused: